metadrive/manager/traffic_manager.py

```diff
@@ -102,17 +102,23 @@ def after_step(self, *args, **kwargs):
                 if self.mode == TrafficMode.Trigger:
                     v_to_remove.append(v)
                 elif self.mode == TrafficMode.Respawn or self.mode == TrafficMode.Hybrid:
-                    lane = self.respawn_lanes[self.np_random.randint(0, len(self.respawn_lanes))]
-                    lane_idx = lane.index
-                    long = self.np_random.rand() * lane.length / 2
-                    v.update_config({"spawn_lane_index": lane_idx, "spawn_longitude": long})
-                    v.reset(random_seed=v.random_seed)
-                    self.engine.get_policy(v.id).reset()
+                    v_to_remove.append(v)
                 else:
                     raise ValueError("Traffic mode error: {}".format(self.mode))
         for v in v_to_remove:
+            vehicle_type = type(v)
             self.clear_objects([v.id])
             self._traffic_vehicles.remove(v)
+            if self.mode == TrafficMode.Respawn or self.mode == TrafficMode.Hybrid:
+                lane = self.respawn_lanes[self.np_random.randint(0, len(self.respawn_lanes))]
+                lane_idx = lane.index
+                long = self.np_random.rand() * lane.length / 2
+                traffic_v_config = {"spawn_lane_index": lane_idx, "spawn_longitude": long}
+                new_v = self.spawn_object(vehicle_type, vehicle_config=traffic_v_config)
```

Comment:
I think here will lose all information in original vehicle config. This is very unsafe here to set "traffic_v_config".

Reply:
I will apply ```copy()``` for ```vehicle_config``` when intialize  ```BaseVehicle```

Reply:
How about storing a copy of the global_config dict, when creating an environment? So that we can compare the one in ``engine``` and this copied version to test if any thing is changed? This can happen in each time when calling reset()

Reply:
question remain